### PR TITLE
(feat) : added billing tabs and search option for patient bills

### PR DESCRIPTION
**Pull request by munania**
Uploading BILLING TABS AND SEARCH.mp4…

## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by munania** [comment]
Uploading BILLING TABS AND SEARCH.mp4…

File: packages/esm-billing-app/src/past-patient-bills/patient-bills-dashboard/patient-bills-dashboard.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@use '@carbon/colors';
+@use '@carbon/layout';
+@use '@carbon/type';
+
+.container {
+  height: calc(100vh - 3rem);
+}
+
+.billsTableContainer {
+  margin: 2rem 1rem;
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/past-patient-bills/patient-bills-dashboard/patient-bills-dashboard.scss:10]:
```suggestion
  margin: layout.$spacing-03; layout.$spacing-01;
```

File: packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+              <p className={styles.content}>{t('noBillDisplay', 'There are no bills to display for this patient')}</p>
+            </Tile>
+          </Layer>
+        </div>
+      </>
+    );
+  }
+
+  return (
+    <div style={{ marginTop: '1rem' }}>
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx:75]:
```suggestion
    <div className={styles.container}>
```
prefer the styles in css

File: packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        {bill.dateCreated}
+      </ConfigurableLink>
+    ),
+    billableService: extractString(bill.billingService),
+    totalAmount: convertToCurrency(bill.totalAmount),
+  }));
+
+  if (bills.length === 0 && patientUuid !== '') {
+    return (
+      <>
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx:59]: use EmptyState component here from either `@openmrs/esm-patient-common-lib` or from `@openmrs/esm-framework` the later is preferred.

File: packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  bills: Array<MappedBill>;
+  setPatientUuid: (patientUuid: string) => void;
+};
+
+const PastPatientBills: React.FC<PatientBillsProps> = ({ patientUuid, bills, setPatientUuid }) => {
+  const { t } = useTranslation();
+
+  if (!patientUuid) {
+    return;
+  }
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx:33]: Its better to have UI showing some information for the component to work is missing e.g 

```tsx
<h3>Missing patient information</h3>
```

**Review by donaldkibet — CHANGES_REQUESTED** [review]
Thanks @munania, the feature is coming up well. Please provide a screen recording of how the feature is. 🚀

**Review by munania — COMMENTED** [review]
UI FIXES DONE

File: packages/esm-billing-app/src/billing-tabs/billling-tabs.component.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    setActiveTabIndex(selectedIndex);
+  };
+
+  return (
+    <div className={styles.referralsList} data-testid="BillingsList-list">
+      <Tabs selectedIndex={activeTabIndex} onChange={handleTabChange} className={styles.tabs}>
+        <div style={{ display: 'flex' }}>
+          <TabList style={{ paddingLeft: '1rem' }} aria-label="Billing tabs" contained>
+            <Tab className={styles.tab}>{"Today's bills"}</Tab>
+            <Tab className={styles.tab}>{'All Bills'}</Tab>
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billing-tabs/billling-tabs.component.tsx:22]:
```suggestion
            <Tab className={styles.tab}>{t('patientBills','Patient Bill')}</Tab>
```

**Review by donaldkibet — APPROVED** [review]

File: packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+import { ConfigurableLink } from '@openmrs/esm-framework';
+import { EmptyState } from '@openmrs/esm-patient-common-lib';
+
+type PatientBillsProps = {
+  patientUuid: string;
+  bills: Array<MappedBill>;
+  setPatientUuid: (patientUuid: string) => void;
+};
+
+const PastPatientBills: React.FC<PatientBillsProps> = ({ patientUuid, bills, setPatientUuid }) => {
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/past-patient-bills/filtered-patient-bills.component.tsx:28]:
```suggestion
const PatientBills: React.FC<PatientBillsProps> = ({ patientUuid, bills, setPatientUuid }) => {
```

**Review by donaldkibet — APPROVED** [review]
Thanks @munania LGTM 🤝, just do a follow up in the last suggestion.

**Pull request merged**